pkg/state/state_nv_ipam_cni_test.go

```diff
@@ -21,33 +21,266 @@ import (
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDeploymentFields(d, &cr.Spec.NvIpam.ImageSpec)
+			d.Status = appsv1.DeploymentStatus{
+				ObservedGeneration: 1,
+				Replicas:           1,
+				UpdatedReplicas:    1,
+				AvailableReplicas:  1,
+				ReadyReplicas:      1,
+			}
+			By("Update Deployment Status, and re-run Sync")
+			err = client.Status().Update(context.Background(), d)
+			Expect(err).NotTo(HaveOccurred())
+			By("Verify State is ready")
+			ctx := context.Background()
+			objs, err := renderer.GetManifestObjects(ctx, cr, catalog, log.FromContext(ctx))
+			Expect(err).NotTo(HaveOccurred())
+			status, err = getKindState(ctx, client, objs, "Deployment")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(BeEquivalentTo(state.SyncStateReady))
+		})
+
+		It("should fail if static config provider not set in catalog", func() {
+			By("Sync")
+			catalog := state.NewInfoCatalog()
+			cr := getMinimalNicClusterPolicyWithNVIpam("nv-ipam-node")
+			status, err := nvIpamState.Sync(context.Background(), cr, catalog)
+			Expect(err).To(HaveOccurred())
+			Expect(status).To(BeEquivalentTo(state.SyncStateError))
+		})
+
+		It("should fail if clustertype provider not set in catalog", func() {
+			By("Sync")
+			catalog := state.NewInfoCatalog()
+			staticConfigProvider := staticconfig_mocks.Provider{}
+			staticConfigProvider.On("GetStaticConfig").Return(staticconfig.StaticConfig{CniBinDirectory: ""})
+			catalog.Add(state.InfoTypeStaticConfig, &staticConfigProvider)
+			cr := getMinimalNicClusterPolicyWithNVIpam("nv-ipam-node")
+			status, err := nvIpamState.Sync(context.Background(), cr, catalog)
+			Expect(err).To(HaveOccurred())
+			Expect(status).To(BeEquivalentTo(state.SyncStateError))
+		})
+	})
+})
+
+func assertValidatingWebhookFields(adm *admv1.ValidatingWebhookConfiguration) {
+	Expect(adm.Webhooks[0].Name).To(Equal("validate-ippool.nv-ipam.nvidia.com"))
+	Expect(adm.Webhooks[1].Name).To(Equal("validate-cidrpool.nv-ipam.nvidia.com"))
+}
+
+func assertPodTemplateWebhookFields(template *v1.PodTemplateSpec) {
+	Expect(template.Spec.Containers[0].Args).Should(ContainElement(ContainSubstring("--webhook=true")))
+	Expect(template.Spec.Volumes).To(Equal(
+		[]v1.Volume{
+			{
+				Name: "cert",
+				VolumeSource: v1.VolumeSource{
+					Secret: &v1.SecretVolumeSource{
+						SecretName:  "nv-ipam-webhook-server-cert",
+						DefaultMode: ptr.To(int32(420)),
+					},
+				},
+			},
+		},
+	))
+
+	Expect(template.Spec.Containers[0].VolumeMounts).To(Equal(
+		[]v1.VolumeMount{
+			{
+				Name:      "cert",
+				MountPath: "/tmp/k8s-webhook-server/serving-certs",
+				ReadOnly:  true,
+			},
+		},
+	))
+}
+
+func getMinimalNicClusterPolicyWithNVIpam(name string) *mellanoxv1alpha1.NicClusterPolicy {
 	cr := getTestClusterPolicyWithBaseFields()
-	cr.Spec.NvIpam = &mellanoxv1alpha1.NVIPAMSpec{
-		// TODO(killianmuldoon): Test with the webhook enabled.
+	cr.Name = "nic-cluster-policy"
```

Comment:
Why do we need Name field only for these new tests?

Reply:
right, we don't need it in the tests here.